### PR TITLE
Add flags to pass '--enable-software-rendering' flag to engine in run command

### PR DESCRIPTION
**Pull request by GaryQian**

### CODE REVIEW

File: packages/flutter_tools/lib/src/commands/run.dart [hunk 1]
```diff
@@ -88,6 +88,12 @@ class RunCommand extends RunCommandBase {
         defaultsTo: false,
         negatable: false,
         help: 'Start in a paused mode and wait for a debugger to connect.');
+    argParser.addFlag('enable-software-rendering',
+        defaultsTo: false,
+        negatable: false,
+        help: 'Enable rendering using the Skia software backend. This is useful\n'
+              'when testing Flutter on emulators. By default, Flutter will\n'
+              'attempt to either use OpenGL or Vulkan.');
```
Comment (yjbanov) [inline packages/flutter_tools/lib/src/commands/run.dart]: nit: suggest ending the last sentence with "and will fallback to software rendering when neither are available."

**Review by yjbanov — APPROVED** [review]

File: packages/flutter_tools/lib/src/commands/run.dart [hunk 2]
```diff
@@ -88,6 +88,12 @@ class RunCommand extends RunCommandBase {
         defaultsTo: false,
         negatable: false,
         help: 'Start in a paused mode and wait for a debugger to connect.');
+    argParser.addFlag('enable-software-rendering',
```
Comment (yjbanov) [inline packages/flutter_tools/lib/src/commands/run.dart]: Are these options shared with `flutter drive` command? If not, could you add it there too?

**Review by yjbanov — APPROVED** [review]

File: packages/flutter_tools/lib/src/android/android_device.dart [hunk 3]
```diff
@@ -397,6 +398,8 @@ class AndroidDevice extends Device {
       cmd.addAll(<String>['--ez', 'trace-startup', 'true']);
     if (route != null)
       cmd.addAll(<String>['--es', 'route', route]);
+    if (enableSoftwareRendering || debuggingOptions.enableSoftwareRendering)
```
Comment (jason-simmons) [inline packages/flutter_tools/lib/src/android/android_device.dart]: Can these two flags be reduced into one?

File: packages/flutter_tools/lib/src/commands/run.dart (hunk 1 above)
Comment (GaryQian) [inline packages/flutter_tools/lib/src/commands/run.dart]: This is not yet the case, but I'll add it in a second!

File: packages/flutter_tools/lib/src/android/android_device.dart (hunk 3 above)
Comment (GaryQian) [inline packages/flutter_tools/lib/src/android/android_device.dart]: Yes!

**Pull request merged**